app/screens/transfer-target-input-screen/transfer-target-input-screen.tsx

```diff
@@ -111,9 +115,23 @@ export class TransferTargetInputScreen extends React.Component<TransferTargetInp
 
   _mapParamsToProps = (prepProps?: TransferTargetInputScreenProps) => {
     const prevAddress = prepProps && prepProps.navigation.getParam("address")
+    const likerId = this.props.navigation.getParam("likerId")
     const address = this.props.navigation.getParam("address")
-    if (!prevAddress && address) {
-      this.props.txStore.setTarget(address)
+    const memo = this.props.navigation.getParam("memo")
+    const amount = this.props.navigation.getParam("amount")
+    const skipToConfirm = this.props.navigation.getParam("skipToConfirm")
+    const target = likerId || address
+    if (!prevAddress && target) {
+      this.props.txStore.setTarget(target)
+    }
+    if (memo) this.props.txStore.setMemo(memo)
```

Comment:
I suspect the loop point is this

`setMemo()` -> `componentDidUpdate()` -> `this._mapParamsToProps()` -> `setMemo()`

Reply:
so I should guard `memo !== memo`, or just like others param above?
but I really want to prevent a case where old memo is used instead of a possible new memo
what kind of UX flow should I expect this function to be triggered, and will it be possible that memo will be wrongly reused?

Reply:
Yes I think you should guard like `prevMemo !== memo`

`_mapParamsToProps` will be called in:
- `componentDidMount` - called once for the first time
- `componentDidUpdate` - called when navigating back to the screen (e.g. Clicking back button), or any state updates

Reply:
Should i change the old code from `if (!prevAddress && target)` to `if (prevAddress !== target)` too?

Reply:
I think yes, but I am not sure whether param is available when going back to the screen, so maybe also need to check the existence `target`